.github/workflows/commit-agent.yml

```
@@ -0,0 +1,18 @@
+on:
+  pull_request:
+    branches: [main]
+
+jobs:
+  critique:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Ensure all branches and commit history are fetched
+      - uses: actions/setup-python@v4
```

actionlint (1.7.8)

the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)